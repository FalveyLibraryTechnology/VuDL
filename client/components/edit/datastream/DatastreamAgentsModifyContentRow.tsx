import React from "react";
import IconButton from "@mui/material/IconButton";
import DatastreamAgentsContentRow from "./DatastreamAgentsContentRow";
import RemoveCircle from "@mui/icons-material/RemoveCircle";
import { useEditorContext } from "../../../context/EditorContext";

export interface Agent {
    role: string;
    type: string;
    name: string;
    notes: Array<string>;
}
export interface DatastreamAgentsModifyContentRowProps {
    agent: Agent;
    index: number;
    setHasChanges: (hasChanges: boolean) => void;
}

const DatastreamAgentsModifyContentRow = ({
    agent,
    index,
    setHasChanges,
}: DatastreamAgentsModifyContentRowProps): React.ReactElement => {
    const {
        state: { currentAgents },
        action: { setCurrentAgents },
    } = useEditorContext();
    const removeAgent = () => {
        currentAgents.splice(index, 1);
        setCurrentAgents([...currentAgents]);
        setHasChanges(true);
    };
    const onChangeKey = (key, value) => {
        currentAgents[index][key] = value;
        setCurrentAgents([...currentAgents]);
        setHasChanges(true);
    };

    return (
        <DatastreamAgentsContentRow
            agent={agent}
            additionalControls={
                <IconButton onClick={() => removeAgent()}>
                    <RemoveCircle />
                </IconButton>
            }
            initialExpand={false}
            onRoleChange={(value) => onChangeKey("role", value)}
            onTypeChange={(value) => onChangeKey("type", value)}
            onNameChange={(value) => onChangeKey("name", value)}
            onNotesChange={(value) => onChangeKey("notes", value)}
        />
    );
};

export default DatastreamAgentsModifyContentRow;
