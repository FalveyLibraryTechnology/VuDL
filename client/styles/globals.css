:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
