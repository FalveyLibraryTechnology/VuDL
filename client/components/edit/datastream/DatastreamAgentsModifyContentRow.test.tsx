import React from "react";
import { describe, afterEach, expect, it, jest } from "@jest/globals";
import renderer from "react-test-renderer";
import DatastreamAgentsModifyContentRow from "./DatastreamAgentsModifyContentRow";

let datastreamAgentsContentRowProps;
const mockDatastreamAgentsContentRow = jest.fn();
jest.mock("./DatastreamAgentsContentRow", () => (props) => {
    datastreamAgentsContentRowProps = props;
    mockDatastreamAgentsContentRow(props);
    return "DatastreamAgentsContentRow";
});
const mockUseEditorContext = jest.fn();
jest.mock("../../../context/EditorContext", () => ({
    useEditorContext: () => {
        return mockUseEditorContext();
    },
}));

describe("DatastreamAgentsModifyContentRow", () => {
    let agent;
    let props;
    let editorValues;

    beforeEach(() => {
        agent = {
            role: "test1",
            type: "test2",
            name: "test3",
            notes: ["test4"],
        };
        props = {
            agent,
            index: 1,
            setHasChanges: jest.fn(),
        };
        editorValues = {
            state: { currentAgents: ["test1"] },
            action: { setCurrentAgents: jest.fn() },
        };
        mockUseEditorContext.mockReturnValue(editorValues);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("proxies DatastreamAgentsContextRow", () => {
        const tree = renderer.create(<DatastreamAgentsModifyContentRow {...props} />).toJSON();
        expect(tree).toEqual("DatastreamAgentsContentRow");
        expect(datastreamAgentsContentRowProps.agent).toEqual(props.agent);
    });
});
