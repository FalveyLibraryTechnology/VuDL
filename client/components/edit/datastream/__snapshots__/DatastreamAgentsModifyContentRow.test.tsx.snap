// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatastreamAgentsModifyContentRow renders 1`] = `
<Component
  additionalControls={
    <ForwardRef(IconButton)
      onClick={[Function]}
    >
      <Memo />
    </ForwardRef(IconButton)>
  }
  agent={
    Object {
      "name": "test3",
      "notes": Array [
        "test4",
      ],
      "role": "test1",
      "type": "test2",
    }
  }
  initialExpand={false}
  onNameChange={[Function]}
  onNotesChange={[Function]}
  onRoleChange={[Function]}
  onTypeChange={[Function]}
/>
`;
